types/react/index.d.ts

```diff
@@ -2602,6 +2603,7 @@ declare namespace React {
         rel?: string;
         target?: string;
         type?: string;
+        as?: string;
```

Comment:
Hi @artem-malko, i've tried to use the `as` attribute on a component that extends `AnchorHTMLAttributes`. Something like the following:
```ts
interface MyCustomComponentProps extends AnchorHTMLAttributes<HTMLAnchorElement> {
  as: React.ComponentClass<{}> | React.StatelessComponent<{}> | string;
}
```

But this is not allowed as the `as` attribute is defined on AnchorHTMLAttributes, so i tracked it down to this PR and through that i found the preload standard. https://w3c.github.io/preload/#as-attribute

The standard clearly defines the `as` attribute for the `<link />` element, but i can't find any info on it also being an attribute for the `<a />` element? Could you clarify to me why you also added it here?

Reply:
@johnnyreilly since you merged it i hope you might know more :)

Reply:
Nice thanks @artem-malko, i'll create a PR to remove it :D